src/plugins/dashboard/public/dashboard_container/state/dashboard_container_reducers.ts

```diff
@@ -126,7 +126,10 @@ export const dashboardContainerReducers = {
   },
 
   resetToLastSavedInput: (state: DashboardReduxState) => {
-    state.explicitInput = state.componentState.lastSavedInput;
+    state.explicitInput = {
+      ...state.componentState.lastSavedInput,
+      viewMode: state.explicitInput.viewMode, // keep current view mode when resetting
```

Comment:
It is possible for a `view` view mode to be included in a dashboard's saved input - for example, in the sample dashboards, their imported state is in `view` mode which meant that hitting the "reset" would actually flip them from edit mode into view mode (since this is included as part of the last saved input). This line ensures that view mode is ignored when resetting to the last saved input.